libsolidity/codegen/ArrayUtils.h

```diff
@@ -67,7 +70,7 @@ class ArrayUtils
 	/// Appends a loop that clears a sequence of storage slots of the given type (excluding end).
 	/// Stack pre: end_ref start_ref
 	/// Stack post: end_ref
-	void clearStorageLoop(Type const& _type) const;
+	void clearStorageLoop(TypePointer const& _type) const;
```

Comment:
Which reference counters (and which of the two `const`s) are you talking about? We can take the pointer by value, since we also copy it inside the closure later, but I guess sticking to passing complex objects by const ref most of the time and then copying later is fine, or do you have a strong argument against it?